### PR TITLE
Advanced routes for merchant_items, admin_merchants, and admin_invoices

### PR DESCRIPTION
**Pull request by jaw772**
- Also refactored the update forms to match the form_with model:

### CODE REVIEW

File: app/models/admin.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Admin < ApplicationRecord
```
Comment (LelandCurtis) [inline app/models/admin.rb:1]: Is this required or was this created as part of bug fix exploration? Would love to chat through how you solved this.

File: config/routes.rb [hunk 2]
```diff
@@ -1,22 +1,27 @@
...
-  get '/merchants/:id/items', to: 'merchant_items#index'
-  get '/merchants/:merchant_id/items/:item_id', to: 'merchant_items#show'
-  get '/merchants/:merchant_id/items/:item_id/edit', to: 'merchant_items#edit'
-  patch '/merchants/:merchant_id/items/:item_id', to: 'merchant_items#update'
-
+  # get '/merchants/:id/items', to: 'merchant_items#index'
+  # get '/merchants/:merchant_id/items/:item_id', to: 'merchant_items#show'
+  # get '/merchants/:merchant_id/items/:item_id/edit', to: 'merchant_items#edit'
+  # patch '/merchants/:merchant_id/items/:item_id', to: 'merchant_items#update'
+  resources :merchants do
```
Comment (LelandCurtis) [inline config/routes.rb:7]: should we throw an only: [:index] here to limit the number of merchants routes created?

File: spec/features/admin/merchants/show_spec.rb [hunk 3]
```diff
@@ -14,12 +14,12 @@
...
     expect(page).to have_link("Update Information")
     click_on "Update Information"
-
-    expect(current_path).to eq("/admin/merchants/#{merchant.id}/edit")
-    expect(page).to have_field(:name, with: "Willy's Bakery")
     
-    fill_in 'name', with: "Jimmy's Bakery"
-    click_on 'Submit'
+    expect(current_path).to eq("/admin/merchants/#{merchant.id}/edit")
+    expect(page).to have_field(:merchant_name, with: "Willy's Bakery")
```
Comment (LelandCurtis) [inline spec/features/admin/merchants/show_spec.rb:19]: slick test of the form_with :model autofill

File: spec/features/merchants/items/show_spec.rb [hunk 4]
```diff
@@ -16,15 +16,15 @@
     merchant1 = create(:merchant)
     item1 = create(:item, merchant: merchant1, unit_price: 77777)
     visit "/merchants/#{merchant1.id}/items/#{item1.id}"
-    expect(page).to have_button("Update #{item1.name}")
+    expect(page).to have_link("Update #{item1.name}")
```
Comment (LelandCurtis) [inline spec/features/merchants/items/show_spec.rb:19]: curious, why did you replace all the buttons with links?

**Review by LelandCurtis — APPROVED** [review]
if my questions don't raise any concerns, we can merge.

File: spec/features/admin/merchants/show_spec.rb (hunk 3 above)
Comment (EMielke76) [reply spec/features/admin/merchants/show_spec.rb:19]: Thanks for finding that syntax for the test Josh, super cool!

**Review by EMielke76 — APPROVED** [review]

**Pull request merged**